File: 주영/Greedy/435. Non-overlapping Intervals.js
/**
 * @param {number[][]} intervals
 * @return {number}
 */
var eraseOverlapIntervals = function (intervals) {
  intervals.sort((a, b) => a[1] - b[1]);

  let cnt = 0;
  let end = intervals[0][1];

  for (let i = 1; i < intervals.length; i++) {
    if (end > intervals[i][0]) cnt++;
    else end = intervals[i][1];
  }
  return cnt;
};

let intervals = [
  [1, 100],
  [11, 22],
  [1, 11],
  [2, 12],
];

eraseOverlapIntervals(intervals);

File: P/435_Non-overlapping_Intervals.java
class Solution {
    public int eraseOverlapIntervals(int[][] intervals) {
        Arrays.sort(intervals, (a, b) -> Integer.compare(a[0], b[0]));
        
        int count = 0;
        for(int i = 1; i < intervals.length; i++) {
            if(intervals[i - 1][1] > intervals[i][0]) {
                count++;
                intervals[i][1] = Math.min(intervals[i - 1][1], intervals[i][1]);
            }
        }
        return count;
    }
}